crates/reth/statediff/src/state.rs

```diff
@@ -0,0 +1,126 @@
+use std::collections::HashMap;
+
+use revm_primitives::{alloy_primitives::Address, B256, U256};
+use strata_mpt::{keccak, MptNode, StateAccount, EMPTY_ROOT};
+
+use crate::BatchStateDiff;
+
+/// An (in-memory) representation of the EVM state reconstructed only from [`BatchStateDiff`].
+#[derive(Clone, Default, Debug)]
+pub struct StateReconstructed {
+    state_trie: MptNode,
+    storage_trie: HashMap<Address, MptNode>,
+}
+
+use thiserror::Error as ThisError;
+
+#[derive(ThisError, Debug)]
+/// A concrete error that may happen during state reconstruction.
+pub enum StateError {
+    #[error("Mpt Error: {0}")]
+    MptError(#[from] strata_mpt::Error),
```

Comment:
In the main strata repo we've been doing this as

```
/// docs
#[derive(Debug, ThisError)]
pub enum StateError {
    #[error("mpt: {0}")]
    ...
```

not sure what the proper style in here *ought* be.

Reply:
Sure, changed for consistency.

Reply:
@storopoli Well, the original comment from Trey was more about whether to put comment before `#[derive]` or after, it's **not** about error handling in IIUC.

Reply:
Yeah I was probably very tired. This is obvious. Sorry about this.
My original intention was to give awareness to the fact that we don't have a proper guidelines on how to do error structs, enums etc with `thiserror` in our guidelines.
I am resolving this for now.